### PR TITLE
tests: posix: common: separate posix xsi system logging tests into a standalone test

### LINKED ISSUE #80986: tests: posix: common: split syslog tests into standalone test
**Is your enhancement proposal related to a problem? Please describe.**
Make the syslog-related tests in `tests/posix/common` into a standalone test.

**Describe the solution you'd like**
The obvious way to group syslog-related testsuites is by `XSI_SYSTEM_LOGGING` [Option Group](https://pubs.opengroup.org/onlinepubs/9799919799/xrat/V4_subprofiles.html).

**Describe alternatives you've considered**
The way the testsuite has been designed historically.

**Additional context**
See #79454 for more info

### PR DESCRIPTION
**Pull request by Pancakem**
Fixes #80986

posix.common contains testsuites that can be separated into smaller groups of tests. This change moves syslog into a singular testsuite at tests/posix/xsi_system_logging app directory.

### CODE REVIEW

File: tests/posix/syslog/prj.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CONFIG_POSIX_API=y
+CONFIG_ZTEST=y
+
+CONFIG_POSIX_AEP_CHOICE_PSE52=y
```
Comment (cfriedt) [inline tests/posix/syslog/prj.conf:4]: Probably only the base system interfaces are needed here.

```suggestion
CONFIG_POSIX_AEP_CHOICE_BASE=y
```

See
https://docs.zephyrproject.org/latest/services/portability/posix/overview/index.html#configuration

**Review by cfriedt — COMMENTED** [review]
Probably only `CONFIG_POSIX_AEP_CHOICE_BASE=y` is needed instead of PSE52.

File: tests/posix/syslog/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(posix_syslog)
+
+FILE(GLOB app_sources src/*.c)
+zephyr_include_directories(${ZEPHYR_BASE}/lib/posix)
```
Comment (cfriedt) [inline tests/posix/syslog/CMakeLists.txt:8]: Can probably be removed
```suggestion
```

File: tests/posix/syslog/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(posix_syslog)
```
Comment (cfriedt) [inline tests/posix/syslog/CMakeLists.txt:5]:
```suggestion
project(xsi_system_logging)
```

**Review by cfriedt — CHANGES_REQUESTED** [review]

File: tests/posix/syslog/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+FILE(GLOB app_sources src/*.c)
+zephyr_include_directories(${ZEPHYR_BASE}/lib/posix)
+
+target_sources(app PRIVATE ${app_sources})
```
Comment (cfriedt) [inline tests/posix/syslog/CMakeLists.txt:10]:
```suggestion
target_sources(app PRIVATE src/main.c)
```

File: tests/posix/syslog/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(posix_syslog)
+
+FILE(GLOB app_sources src/*.c)
```
Comment (cfriedt) [inline tests/posix/syslog/CMakeLists.txt:7]: This can be removed
```suggestion
```

File: tests/posix/syslog/testcase.yaml
Comment (cfriedt) [inline tests/posix/syslog/testcase.yaml:1]: Please rename the `syslog` directory to `xsi_system_logging`

File: tests/posix/xsi_system_logging/testcase.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  tags:
+    - xsi.system.logging
+  # 1 tier0 platform per supported architecture
+  platform_key:
+    - arch
+    - simulation
+tests:
+  portability.xsi.system.logging:
+    min_flash: 64
+    min_ram: 32
```
Comment (cfriedt) [inline tests/posix/xsi_system_logging/testcase.yaml:12]: This looks like the same failure signature as the other testsuite, when testing newlib on a platform with only 16 kB of SRAM. 

```suggestion
    - simulation
  min_flash: 64
  min_ram: 32
tests:
  portability.xsi.system.logging:
```

**Review by cfriedt — APPROVED** [review]

**Review by ycsin — APPROVED** [review]

**Pull request merged**